### PR TITLE
support labelSelector in restAPI

### PR DESCRIPTION
**Pull request by Binyang2014**

### CODE REVIEW

File: src/rest-server/src/models/kubernetes/kubernetes.js [hunk 1]
```diff
@@ -95,15 +96,12 @@ const getNodes = async () => {
...
   const client = getClient();
-  const labelSelectorStr = encodeSelector(labelSelector, negativeLabelSelector);
+
   const requestOptions = {};
-  if (labelSelectorStr) {
-    requestOptions.params = {
-      labelSelector: labelSelectorStr,
-    };
-  }
+  requestOptions.params = cloneDeep(options);
```
Comment (abuccts) [inline src/rest-server/src/models/kubernetes/kubernetes.js]: in es6, you could use built-in `Object.assign()`

File: src/rest-server/src/models/kubernetes/kubernetes.js [hunk 2]
```diff
@@ -95,15 +96,12 @@ const getNodes = async () => {
...
+  const {namespace} = options;
   const client = getClient();
-  const labelSelectorStr = encodeSelector(labelSelector, negativeLabelSelector);
+
   const requestOptions = {};
-  if (labelSelectorStr) {
-    requestOptions.params = {
-      labelSelector: labelSelectorStr,
-    };
-  }
```
Comment (abuccts) [inline src/rest-server/src/models/kubernetes/kubernetes.js]: seems you can use `options` object directly if you do not need to generate `labelSelectorStr` anymore

File: src/rest-server/src/models/kubernetes/kubernetes.js [hunk 3]
```diff
@@ -99,14 +98,11 @@ const getPods = async (options = {}) => {
   const {namespace} = options;
   const client = getClient();
 
-  const requestOptions = {};
-  requestOptions.params = cloneDeep(options);
-  delete requestOptions.params.namespace;
```
Comment (abuccts) [inline src/rest-server/src/models/kubernetes/kubernetes.js]: do you need to `delete options.namespace`?

Comment (Binyang2014) [reply src/rest-server/src/models/kubernetes/kubernetes.js]: k8s API server will ignore unknown query parameter. But I don't find any doc for this. Remove `options.namespace` is a more safety way.

I don't want to change the input object, will clone a new object and remote `namespace` property.

**Comment by Binyang2014** [comment]
Closes #4252

**Pull request merged**